src/github/mod.rs

```diff
@@ -182,6 +186,218 @@ impl SyncGitHub {
         }))
     }
 
+    fn diff_repos(&self) -> Result<Vec<RepoDiff>, Error> {
+        let mut diffs = Vec::new();
+        for repo in &self.repos {
+            diffs.push(self.diff_repo(repo)?);
+        }
+        Ok(diffs)
+    }
+
+    fn diff_repo(&self, expected_repo: &rust_team_data::v1::Repo) -> Result<RepoDiff, Error> {
+        let actual_repo = match self.github.repo(&expected_repo.org, &expected_repo.name)? {
+            Some(r) => r,
+            None => {
```

Comment:
Hmm, I think we can simplify this function by emitting multiple diffs when a new repo needs to be created. We could have the `None` arm emit a `CreateRepo` diff and return a fake (and empty) `actual_repo`, so that the following code will also emit the relevant update diffs to setup branch protection and permissions.

This would mean `diff_repo` returns a `Vec<RepoDiff>`, which also allows to flatten the `RepoDiff` enum and the variants nested within `RepoDiff::Update`.

Reply:
See my response for your other comment: https://github.com/rust-lang/sync-team/pull/26#discussion_r1043361606

Reply:
This `None` block now shares its diffing logic with the `RepoDiff::Update` flow.